src/tools/sssctl/sssctl_logs.c

```diff
@@ -387,6 +384,10 @@ errno_t sssctl_debug_level(struct sss_cmdline *cmdline,
 
 fini:
     talloc_free(ctx);
+    /* pc_config_file is allocated by popt using malloc().
+     * debug_as_string is not allocated but points to the command line. */
```

Comment:
Why?
Isn't it https://github.com/rpm-software-management/popt/blob/7182e4618ad5a0186145fc2aa4a98c2229afdfa8/src/popt.c#L1277 ?

Reply:
I really can't tell.

I initially freed it and the application was crashing. After investigation I found the returned pointer was pointing to the ``argv[]`` element.

While writing this message, I made a new try. I modified the code to look like this:
```
fini:
    talloc_free(ctx);
    if (cmdline->argv[0] == debug_as_string) {
        DEBUG(SSSDBG_CRIT_FAILURE, "debug_as_string points to argv\n");
    } else {
        DEBUG(SSSDBG_CRIT_FAILURE, "debug_as_string does not point to argv\n");
    }
    /* debug_as_string is not allocated but points to the command line. */
    free(discard_const(debug_as_string));
    return ret;
```
and I got this:
```
$ sudo .libs/sssctl --debug 8 debug-level 0x0070
...
[sssd] [sssctl_debug_level] (0x0020): debug_as_string points to argv
munmap_chunk(): invalid pointer
Aborted
```
Confirming what I initially saw.
Running it without the parameter:
```
$ sudo .libs/sssctl --debug 8 debug-level 0x0070
...
[sssd] [sssctl_debug_level] (0x0020): debug_as_string does not point to argv
```
Has the expected behavior.

I tried a quick debugging session but the function returned almost immediately, probably because of some compiler optimization. If you think sorting this out is worth the effort, I am ready to investigate what is happening.

Reply:
Probably not worth, taking into account this is a process termination anyway.